packages/rstream/src/api.ts

```diff
@@ -94,53 +107,86 @@ export interface ISubscriber<T> {
+    /**
+     * Adds given `sub` as child subscription.
+     *
+     * @param sub
+     */
+    subscribe<B>(sub: Subscription<A, B>): Subscription<A, B>;
+    /**
+     * Wraps given partial `sub` in a {@link Subscription} and attaches it as
+     * child subscription.
+     *
+     * @param sub
+     * @param opts
+     */
     subscribe(
-        sub: Partial<ISubscriber<T>>,
-        opts?: Partial<CommonOpts>
-    ): Subscription<T, T>;
-    subscribe<C>(
-        sub: Partial<ISubscriber<T>>,
-        xform: Transducer<T, C>,
-        opts?: Partial<CommonOpts>
-    ): Subscription<T, C>;
-    subscribe<C>(
-        xform: Transducer<T, C>,
+        sub: Partial<ISubscriber<A>>,
         opts?: Partial<CommonOpts>
-    ): Subscription<T, C>;
-    subscribe<C>(sub: Subscription<T, C>): Subscription<T, C>;
-    unsubscribe(sub?: Partial<ISubscriber<T>>): boolean;
+    ): Subscription<A, A>;
+    /**
+     * Wraps given partial `sub` in a {@link Subscription} and attaches it as
+     * child subscription. If `opts` defines a transducer (via `xform` key),
+     * input values will be transformed first before reaching the child sub's
+     * {@link ISubsciber.next} handler. Any further downstream subscriptions
+     * attached to the returned wrapped sub will also only receive those
+     * transformed values.
+     *
+     * @see {@link ITransformable}
+     *
+     * @param sub
+     * @param opts
+     */
+    subscribe<B>(
+        sub: Partial<ISubscriber<B>>,
+        opts?: Partial<TransformableOpts<A, B>>
+    ): Subscription<A, B>;
+    /**
+     * Removes given child sub, or if `sub` is omitted, detaches this
+     * subscription itself from its upstream parent (possibly triggering a
+     * cascade of further unsubscriptions, depending on
+     * {@link CommonOpts.closeOut} settings of parent(s)).
+     *
+     * @param sub
+     */
+    unsubscribe(sub?: Partial<ISubscriber<A>>): boolean;
     getState(): State;
 }
 
 export interface ITransformable<B> {
     transform<C>(
         a: Transducer<B, C>,
-        opts?: Partial<CommonOpts>
+        opts?: Partial<WithErrorHandlerOpts>
     ): Subscription<B, C>;
     transform<C, D>(
         a: Transducer<B, C>,
         b: Transducer<C, D>,
-        opts?: Partial<CommonOpts>
+        opts?: Partial<WithErrorHandlerOpts>
     ): Subscription<B, D>;
     transform<C, D, E>(
         a: Transducer<B, C>,
         b: Transducer<C, D>,
         c: Transducer<D, E>,
-        opts?: Partial<CommonOpts>
+        opts?: Partial<WithErrorHandlerOpts>
     ): Subscription<B, E>;
     transform<C, D, E, F>(
         a: Transducer<B, C>,
         b: Transducer<C, D>,
         c: Transducer<D, E>,
         d: Transducer<E, F>,
-        opts?: Partial<CommonOpts>
+        opts?: Partial<WithErrorHandlerOpts>
     ): Subscription<B, F>;
+    transform<C>(
+        opts: WithTransform<B, C> & Partial<WithErrorHandlerOpts>
+    ): Subscription<B, C>;
```

Comment:
This override addresses @jamieowen's comment & use case discussed here: https://github.com/thi-ng/umbrella/issues/276#issuecomment-791288860